### PR TITLE
Update build tools

### PR DESCRIPTION
**Pull request by vincentfretin**
Update to
- Babel 7
- Jest 23
- Webpack 4
- extract-text-webpack-plugin is not working with webpack 4, it has been replaced by mini-css-extract-plugin
- json-loader is not needed since webpack 2 actually, so I removed it
- I simplified webpack config with  `mode: 'production'/'development'` (`new UglifyJSPlugin({ sourceMap: true, parallel: true, cache: true })` is used by default in production mode
- rename .babelrc to babel.config.js, see https://github.com/facebook/jest/issues/6053#issuecomment-383632515

develop branch:
first build (without uglify cache): 1m54.120s
second build:  0m54.846s
without uglify: 0m38.620s

4M, 4102630 bytes with uglify
10M, 10410066 bytes without uglify

babel7 branch:
first build (without uglify cache): 1m58.283s
second build 0m59.051s
without uglify: 0m44.785s
3,9M 4033232 bytes with uglify
8,5M, 8886172 bytes without uglify

### CODE REVIEW

File: assembl/static2/yarn.lock [hunk 1]
```diff
@@ -5,10 +5,12 @@
 "@2fd/command@^1.1.2":
   version "1.1.2"
   resolved "https://registry.yarnpkg.com/@2fd/command/-/command-1.1.2.tgz#b831d511bdd3b4e2d88139aaddb06d3d6aa0b08a"
+  integrity sha1-uDHVEb3TtOLYgTmq3bBtPWqgsIo=
```
Comment (ayyazdaniaryan) [inline assembl/static2/yarn.lock]: Could you please comment how you added the integrity field to yarn (along with why?).

Comment (vincentfretin) [reply assembl/static2/yarn.lock]: just execute yarn with latest version :)

File: assembl/static2/webpack.config.js [hunk 2]
```diff
@@ -60,14 +60,14 @@ module.exports = {
                 // A dependency is transpiled only if it's in the include below.
                 babelrc: false,
                 plugins: [
-                  'transform-object-rest-spread',
-                  'transform-class-properties',
-                  ['transform-runtime', { helpers: true, polyfill: false }]
+                  '@babel/plugin-proposal-object-rest-spread',
+                  '@babel/plugin-proposal-class-properties',
+                  ['@babel/plugin-transform-runtime', { helpers: true }]
```
[the remark this answers is not in the document]
Comment (vincentfretin) [reply assembl/static2/webpack.config.js]: The polyfill option was removed https://babeljs.io/docs/en/babel-plugin-transform-runtime
We still include `@babel/polyfill` in `index.jsx`. 
Here we use transform-runtime with `helpers: true` to reduce the bundle of some kB.

File: assembl/static2/webpack.config.js [hunk 3]
```diff
@@ -96,10 +98,6 @@ module.exports = {
           exclude: /node_modules/,
           use: 'graphql-tag/loader'
         },
-        {
```
Comment (cedricmessiant) [inline assembl/static2/webpack.config.js]: why don't we need this anymore?

Comment (vincentfretin) [reply assembl/static2/webpack.config.js]: json-loader ? webpack handles json import natively since webpack 2

**Comment by vincentfretin** [comment]
I can rebase and merge when the other big PRs are merged and no big branches are pending, in a few weeks. There is no hurry.

File: assembl/static2/webpack.config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ayyazdaniaryan) [reply assembl/static2/webpack.config.js]: Nice, good answer. Thank you, it's more clear now.

**Comment by vincentfretin** [comment]
I had to update storybook to latest 4.0.0-rc.0 which use webpack 4 and babel 7 to fix tests issue after the rebase. BUT storybook 4 now needs node >= 8.0.0, you can't run `npm run storybook` with the node 6 currently included in the venv. So this PR is blocked until we update node in the venv I'm afraid.

**Pull request merged**